front/src/modules/trainschedule/components/ImportTimetableItem/ImportTimetableItemConfig.tsx

```
@@ -275,12 +298,159 @@ const ImportTimetableItemConfig = ({
+        name: step.name,
+        relativeArrivalSeconds: Math.round(
+          (new Date(step.arrivalTime).getTime() - departureTime) / 1000
+        ),
+        relativeDepartureSeconds: Math.round(
+          (new Date(step.departureTime).getTime() - departureTime) / 1000
+        ),
+      }));
+
+      return JSON.stringify(stepsWithRelativeTimesAndNames);
+    };
+
+    const findMostFrequentScheduleInMission = (schedules: ImportedTrainSchedule[]) => {
+      const scheduleOccurrences: Record<
+        string,
+        { count: number; schedule: ImportedTrainSchedule }
+      > = {};
+
+      schedules.forEach((schedule) => {
+        const relativeSteps = getRelativeStepsTimeAndNames(schedule);
+
+        if (scheduleOccurrences[relativeSteps]) {
+          scheduleOccurrences[relativeSteps].count += 1;
+        } else {
+          scheduleOccurrences[relativeSteps] = { count: 1, schedule };
+        }
+      });
+
+      let mostFrequent: ImportedTrainSchedule | null = null;
+      let highestCount = 0;
+
+      Object.values(scheduleOccurrences).forEach(({ count, schedule }) => {
+        if (count > highestCount) {
+          highestCount = count;
+          mostFrequent = schedule;
+        }
+      });
+
+      return { mostFrequent, highestCount };
+    };
+
+    const missionMostFrequentSchedules: Record<
+      string,
+      { schedule: ImportedTrainSchedule | null; count: number }
+    > = {};
+
+    Object.entries(missions).forEach(([missionId, schedules]) => {
+      const { mostFrequent, highestCount } = findMostFrequentScheduleInMission(schedules);
+      missionMostFrequentSchedules[missionId] = { schedule: mostFrequent, count: highestCount };
+    });
+
+    const getMostFrequentInterval = (schedules: ImportedTrainSchedule[]): Duration => {
+      const departureTimes = schedules
+        .map((s) => new Date(s.departureTime))
+        .sort((a, b) => a.getTime() - b.getTime());
+
+      const intervalsCount: Record<number, number> = {};
```

`Record` is not great when keys aren't `string`s, because keys are silently converted to numbers. `Map` may be better suited here.